libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -320,28 +318,24 @@ void DataFlowAnalyzer::clearValues(set<YulString> _variables)
 	// First clear storage knowledge, because we do not have to clear
 	// storage knowledge of variables whose expression has changed,
 	// since the value is still unchanged.
-	for (auto const& name: _variables)
-	{
-		// clear slot denoted by "name"
-		m_storage.eraseKey(name);
-		// clear slot contents denoted by "name"
-		m_storage.eraseValue(name);
-		// assignment to slot denoted by "name"
-		m_memory.eraseKey(name);
-		// assignment to slot contents denoted by "name"
-		m_memory.eraseValue(name);
-	}
+	auto eraseCondition = [&](auto const& entry) {
+		return _variables.count(entry.first) || _variables.count(entry.second);
+	};
+	cxx20::erase_if(m_storage, eraseCondition);
+	cxx20::erase_if(m_memory, eraseCondition);
 
 	// Also clear variables that reference variables to be cleared.
 	for (auto const& name: _variables)
-		for (auto const& ref: m_references.backward[name])
-			_variables.emplace(ref);
+		for (auto const& [ref, names]: m_references)
+			if (names.count(name))
```

Comment:
`names` can now contain variables hat have gone out of scope, correct?

Reply:
Yep